torch/csrc/cuda/Stream.cpp

```diff
@@ -21,13 +21,23 @@ static PyObject* THCPStream_pynew(
   const auto current_device = c10::cuda::current_device();
 
   int priority = 0;
-  uint64_t cdata = 0;
+  int64_t stream_id = 0;
+  int64_t device_index = 0;
+  int64_t device_type = static_cast<int64_t>(c10::DeviceType::CUDA);
   uint64_t stream_ptr = 0;
 
   // NOLINTNEXTLINE(modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-  static char* kwlist[] = {"priority", "_cdata", "stream_ptr", nullptr};
+  static char* kwlist[] = {
+      "priority", "_stream_id", "_device_index", "stream_ptr", nullptr};
   if (!PyArg_ParseTupleAndKeywords(
-          args, kwargs, "|iKK", kwlist, &priority, &cdata, &stream_ptr)) {
+          args,
```

Comment:
Similar to the previous case, it seems that this API is user-accessible? TODO: attempt to match behavior from passing a single `int64_t` as much as possible.